tf_encrypted/protocol/pond/pond.py

```diff
@@ -2382,8 +2382,11 @@ def _truncate_private_interactive(
     # assumption that the values originally lie in `[-B, B)`, and will
     # leak private information otherwise
 
+    # 'a + bound' will automatically lift 'bound' by another scaling factor,
+    # so we should first divide bound by the scaling factor if we want to
+    # use this convenient '+' operation.
     bound = prot.fixedpoint_config.bound_double_precision
-    b = a + bound
+    b = a + (bound / scaling_factor)
```

Comment:
fixing it this way would leave the lower bits zero afaict due to the multiplication by the scaling factor. it might be better to avoid using the higher-level operations although and unwrap first thing.